include/States.h

```diff
@@ -23,8 +25,17 @@ namespace ale {
       velocity = {vec[3], vec[4], vec[5]};
     };
 
+    State(Vec3d position) : position(position) {
+      velocity = {std::numeric_limits<double>::quiet_NaN(),
+                  std::numeric_limits<double>::quiet_NaN(),
+                  std::numeric_limits<double>::quiet_NaN()};
+    };
     State(Vec3d position, Vec3d velocity) : position(position), velocity(velocity) {};
     State() {};
+
+    bool hasVelocity() const {
+      return !(std::isnan(velocity.x) && std::isnan(velocity.y) && std::isnan(velocity.z));
```

Comment:
This makes sense literally and logically, but since NaNs kind of infect any calculation they're involved in, is it  useful to have a velocity with 1-2 components, but not 3, that are NaN? So then, ALE will just do the calculations requested, any calculation with a NaN ends up NaN, and that's just the output from ALE This seems reasonable to me, but I wanted to bring it up anyway.

Reply:
There's no reason why you'd have 1-2 NaNs. I could change these to ors instead.

Reply:
Yeah, having 1-2 NaNs seemed like a weird (impossible?) case. I'd be pro-changing them to ors.